E0088 needs to be updated to new format

From: src/test/compile-fail/E0088.rs

Error E0088 needs to be updated with a span_label, taking it from:

```
error[E0088]: too many lifetime parameters provided: expected 0 parameters, found 1 parameter
  --> src/test/compile-fail/E0088.rs:14:9
   |
14 |     f::<'static>(); //~ ERROR E0088
   |         ^^^^^^^
```

To:

```
error[E0088]: too many lifetime parameters provided: expected 0 parameters, found 1 parameter
  --> src/test/compile-fail/E0088.rs:14:9
   |
14 |     f::<'static>(); //~ ERROR E0088
   |         ^^^^^^^ 0 lifetime parameters expected
```

Bonus: we may want to detect the 0 case and say:

```
error[E0088]: too many lifetime parameters provided: expected 0 parameters, found 1 parameter
  --> src/test/compile-fail/E0088.rs:14:9
   |
14 |     f::<'static>(); //~ ERROR E0088
   |         ^^^^^^^ unexpected lifetime parameter
```

--- a/src/librustc_typeck/check/mod.rs
+++ b/src/librustc_typeck/check/mod.rs
@@ -4412,13 +4412,7 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
         // Check provided lifetime parameters.
         let lifetime_defs = segment.map_or(&[][..], |(_, generics)| &generics.regions);
         if lifetimes.len() > lifetime_defs.len() {
-            let span = lifetimes[..].into_iter().skip(1).map(|lft| lft.span)
-                .fold(lifetimes[0].span, |acc, n| Span {
-                    expn_id: acc.expn_id,
-                    lo: acc.lo,
-                    hi: n.hi,
-                });
-
+            let span = lifetimes[lifetime_defs.len()].span;
             struct_span_err!(self.tcx.sess, span, E0088,
                              "too many lifetime parameters provided: \
                               expected {}, found {}",
@@ -4430,7 +4424,7 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
         }
 
         // The case where there is not enough lifetime parameters is not checked,
-        // because this is not possible - a function never takes lifetime parameters. 
+        // because this is not possible - a function never takes lifetime parameters.
         // See discussion for Pull Request 36208.
 
         // Check provided type parameters.

--- a/src/librustc_typeck/check/mod.rs
+++ b/src/librustc_typeck/check/mod.rs
@@ -4412,19 +4412,26 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
         // Check provided lifetime parameters.
         let lifetime_defs = segment.map_or(&[][..], |(_, generics)| &generics.regions);
         if lifetimes.len() > lifetime_defs.len() {
-            let span = lifetimes[lifetime_defs.len()].span;
-            span_err!(self.tcx.sess, span, E0088,
-                      "too many lifetime parameters provided: \
-                       expected {}, found {}",
-                      count(lifetime_defs.len()),
-                      count(lifetimes.len()));
-        } else if lifetimes.len() > 0 && lifetimes.len() < lifetime_defs.len() {
-            span_err!(self.tcx.sess, span, E0090,
-                      "too few lifetime parameters provided: \
-                       expected {}, found {}",
-                      count(lifetime_defs.len()),
-                      count(lifetimes.len()));
+            let span = lifetimes[..].into_iter().skip(1).map(|lft| lft.span)
+                .fold(lifetimes[0].span, |acc, n| Span {
+                    expn_id: acc.expn_id,
+                    lo: acc.lo,
+                    hi: n.hi,
+                });
+
+            struct_span_err!(self.tcx.sess, span, E0088,
+                             "too many lifetime parameters provided: \
+                              expected {}, found {}",
+                              count(lifetime_defs.len()),
+                              count(lifetimes.len()))
+                .span_label(span, &format!("unexpected lifetime parameter{}",
+                                           match lifetimes.len() { 1 => "", _ => "s" }))
+                .emit();
         }
+
+        // The case where there is not enough lifetime parameters is not checked,
+        // because this is not possible - a function never takes lifetime parameters. 
+        // See discussion for Pull Request 36208.
 
         // Check provided type parameters.
         let type_defs = segment.map_or(&[][..], |(_, generics)| {

--- a/src/test/compile-fail/E0088.rs
+++ b/src/test/compile-fail/E0088.rs
@@ -14,7 +14,7 @@ fn g<'a>() {}
 fn main() {
     f::<'static>(); //~ ERROR E0088
     //~^ unexpected lifetime parameter
-    
+
     g::<'static, 'static>(); //~ ERROR E0088
     //~^ unexpected lifetime parameters
 }

--- a/src/test/compile-fail/E0088.rs
+++ b/src/test/compile-fail/E0088.rs
@@ -9,7 +9,12 @@
 // except according to those terms.
 
 fn f() {}
+fn g<'a>() {}
 
 fn main() {
     f::<'static>(); //~ ERROR E0088
+    //~^ unexpected lifetime parameter
+    
+    g::<'static, 'static>(); //~ ERROR E0088
+    //~^ unexpected lifetime parameters
 }

